engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/sink/KafkaAvroSinkFactoryWithEditor.scala

```diff
@@ -0,0 +1,110 @@
+  )
+}
+
+class KafkaAvroSinkFactoryWithEditor(val schemaRegistryProvider: SchemaRegistryProvider, val processObjectDependencies: ProcessObjectDependencies)
+  extends SinkFactory with KafkaAvroBaseTransformer[FlinkSink] {
+  import KafkaAvroSinkFactoryWithEditor._
+
+  private var sinkValueParameter: Option[AvroSinkValueParameter] = None
+
+  protected def topicParamStep(implicit nodeId: NodeId): NodeTransformationDefinition = {
+    case TransformationStep(Nil, _) =>
+      val topicParam = getTopicParam.map(List(_))
+      NextParameters(parameters = topicParam.value, errors = topicParam.written)
+  }
+
+  protected def schemaParamStep(implicit nodeId: NodeId): NodeTransformationDefinition = {
+    case TransformationStep((TopicParamName, DefinedEagerParameter(topic: String, _)) :: Nil, _) =>
+      val preparedTopic = prepareTopic(topic)
+      val version = getVersionParam(preparedTopic)
+      NextParameters(version.value :: paramsDeterminedAfterSchema, errors = version.written)
+    case TransformationStep((TopicParamName, _) :: Nil, _) =>
+      NextParameters(parameters = fallbackVersionOptionParam :: paramsDeterminedAfterSchema)
+  }
+
+  private def valueParamStep(context: ValidationContext)(implicit nodeId: NodeId): NodeTransformationDefinition = {
+    case TransformationStep
+      (
+        (TopicParamName, DefinedEagerParameter(topic: String, _)) ::
+        (SchemaVersionParamName, DefinedEagerParameter(version: String, _)) ::
+        (SinkValidationModeParameterName, _) ::
+        (SinkKeyParamName, _) :: Nil, _
+      ) =>
+      val preparedTopic = prepareTopic(topic)
+      val versionOption = parseVersionOption(version)
+      val schemaDeterminer = prepareSchemaDeterminer(preparedTopic, versionOption)
+      val schemaValidated = schemaDeterminer
+        .determineSchemaUsedInTyping
+        .leftMap(SchemaDeterminerErrorHandler.handleSchemaRegistryError(_))
+      schemaValidated.andThen { runtimeSchema =>
+        val typing = AvroSchemaTypeDefinitionExtractor.typeDefinition(runtimeSchema.schema)
+        AvroSinkValueParameter(typing).map { valueParam =>
+          sinkValueParameter = Some(valueParam)
+          NextParameters(valueParam.toParameters)
+        }
+      }.valueOr(e => FinalResults(context, e :: Nil))
+  }
+
+  protected def finalParamStep(context: ValidationContext)(implicit nodeId: NodeId): NodeTransformationDefinition = {
+    case TransformationStep(
+      (TopicParamName, _) :: (SchemaVersionParamName, _) :: (SinkValidationModeParameterName, _) :: (SinkKeyParamName, _) ::
+      valueParams, _) if valueParams.nonEmpty => FinalResults(context, Nil)
+  }
+
+  override def initialParameters: List[Parameter] = {
+    implicit val nodeId: NodeId = NodeId("")
+    val topic = getTopicParam.value
+    topic :: getVersionParam(Nil) :: paramsDeterminedAfterSchema
+  }
+
+  override def contextTransformation(context: ValidationContext, dependencies: List[NodeDependencyValue])
```

Comment:
:+1:  nice idea, now we can extract/share some logic with "raw" sink more easily